src/primitiveGuards.ts

```diff
@@ -16,3 +18,15 @@ export const isBoolean = getPrimitiveTypeCheck<boolean>('boolean');
 export const isFunction = getPrimitiveTypeCheck<Function>('function');
 export const isSymbol = getPrimitiveTypeCheck<Symbol>('symbol');
 export const isBigInt = getPrimitiveTypeCheck<BigInt>('bigint');
+
+export function isStringLiteral<T extends string>(keys: {[P in T]: any}): ReasonGuard<unknown, T> {
```

Comment:
Why does this not use an array/tuple?

Reply:
Because I couldn't figure out away to get the generic types for that working so that you can infer all of the literal values from the input object.